### ISSUE TITLE
EnsureArg.IsNotNull for nullable value type

### ISSUE THREAD

**Issue by hannasm** [danielwertheim/Ensure.That#154]
```csharp
Uri? a = null;
EnsureArg.IsNotNull(a);
var x = a.ToString();
```

On line 3, Visual studio generates a CS8602, dereference of a possibly null reference

**Comment by ndrwrbgs** [comment]
Brute force testing seems the current SA from roslyn team requires EXPLICITLY the https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.notnullattribute?view=net-5.0 attribute, without compatibility with the existing R# attributes.

Sadly, the old `ValidatedNotNull` will still be required for projects using those versions, and `NotNull` will be required for new projects, while `ContractAnnotation` is required for CodeContracts based projects. Microsoft has quite the repertoire of static analysis approaches. I apologize in advance for how attribute-y the code is getting @danielwertheim!

It does appear the nice JetBrains team made compat with the C#8 attributes though, so we can simply replace in some places.

**Comment by hannasm** [comment]
Sure i'd be glad to check it

**Comment by danielwertheim** [comment]
There's now a pre-release out regarding this: https://www.nuget.org/packages/Ensure.That/10.1.0-p1

**Comment by danielwertheim** [comment]
@hannasm did the preview release solve your issue?

**Comment by hannasm** [comment]
I can confirm this issue is resolved for me.

There is a substantial API surface to validate and I only tested a few use
cases.

I was thinking that perhaps you could create a project with 'warnings as
errors' around these use cases to automate the validation. I'm sure
multiple build targets and varying toolchains will complicate that setup,
and it's a bit unconventional.

**Comment by ndrwrbgs** [comment]
The hard part is that most of the impact for other toolchains will be
authoring time only and not a build break. I'd ship it, myself, since the
issues it can cause aren't program validity issues but one of { someone
upgrading's build will outright break, someone's IDE/tools will false warn
about something }.

Neither is ideal, but I see both as low impact to a product and easy to fix
if reported.

**Comment by danielwertheim** [comment]
Thanks @hannasm for verifying.

@ndrwrbgs Agreed. It's now released as `v10.1.0`. Thanks for all the work fixing this.
